src/main/java/org/cbio/gdcpipeline/tasklet/ProcessManifestFileTasklet.java

```diff
@@ -0,0 +1,219 @@
+package org.cbio.gdcpipeline.tasklet;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.ManifestFileData;
+import org.cbio.gdcpipeline.model.rest.response.GdcApiResponse;
+import org.cbio.gdcpipeline.model.rest.response.GdcFileMetadata;
+import org.cbio.gdcpipeline.model.rest.response.Hits;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.batch.item.file.mapping.DefaultLineMapper;
+import org.springframework.batch.item.file.mapping.FieldSetMapper;
+import org.springframework.batch.item.file.transform.DelimitedLineTokenizer;
+import org.springframework.batch.item.file.transform.FieldSet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Dixit Patel
+ */
+public class ProcessManifestFileTasklet implements Tasklet {
+    @Value("#{jobParameters[manifest_file]}")
+    private String manifest_file;
+
+    @Value("${gdc.api.files.endpoint}")
+    private String GDC_API_FILES_ENDPOINT;
+
+    @Value("${gdc.max.response.size}")
+    private int MAX_RESPONSE_SIZE;
+
+    private static Log LOG = LogFactory.getLog(ProcessManifestFileTasklet.class);
+    private List<ManifestFileData> manifestFileList = new ArrayList<>();
+    private RestTemplate restTemplate = new RestTemplate();
+    private List<GdcFileMetadata> gdcFileMetadatas = new ArrayList<>();
+
+    @Override
+    public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+        FlatFileItemReader<ManifestFileData> reader = new FlatFileItemReader<>();
+        DelimitedLineTokenizer tokenizer = new DelimitedLineTokenizer(DelimitedLineTokenizer.DELIMITER_TAB);
+        try (FileReader r = new FileReader(manifest_file)) {
```

Comment:
I think it would be better to instantiate the FileReader on line 58 inside the BufferedReader constructor - it's a more standard design pattern.